### PR TITLE
[WIP] Implement Ghost Squadron

### PR DESCRIPTION
**Pull request by TheMolkaPL**
Original source: git@github.com:OvercastNetwork/ProjectAres.git

### CODE REVIEW

File: src/main/java/tc/oc/pgm/ghostsquadron/GhostSquadronMatchModule.java [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+    this.demoClass = classMatchModule.getPlayerClass("Demo");
+    // this.ninjaClass = checkNotNull(classMatchModule.getPlayerClass("Ninja"), "Ninja class not
+    // found");
+  }
+
+  @Override
+  public void enable() {
+    GhostSquadronTask task = new GhostSquadronTask(this.match, this, this.classMatchModule);
+    this.mainTask = Bukkit.getScheduler().runTaskTimer(PGM.get(), task, 0, 10);
+    this.revealTask = Bukkit.getScheduler().runTaskTimer(PGM.get(), new RevealTask(this), 0, 1);
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/ghostsquadron/GhostSquadronMatchModule.java]: This must be migrated to match-scoped scheduler

File: src/main/java/tc/oc/pgm/ghostsquadron/GhostSquadronMatchModule.java [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+      this.spideySenses.put(userId, now);
+
+      player
+          .getBukkit()
+          .addPotionEffect(new PotionEffect(PotionEffectType.NIGHT_VISION, 4 * 20, 0), true);
+      player.getBukkit().addPotionEffect(new PotionEffect(PotionEffectType.SPEED, 4 * 20, 1), true);
+
+      player.getBukkit().playSound(player.getBukkit().getLocation(), Sound.SPIDER_IDLE, 5, 0);
+      player.getBukkit().playSound(player.getBukkit().getLocation(), Sound.SPIDER_IDLE, 5, 0.25f);
+      player.getBukkit().playSound(player.getBukkit().getLocation(), Sound.SPIDER_IDLE, 5, 0.5f);
```
Comment (TheMolkaPL) [inline src/main/java/tc/oc/pgm/ghostsquadron/GhostSquadronMatchModule.java]: I'm not sure about the `Sound.SPIDER_IDLE`, it was `Sound.ENTITY_SPIDER_AMBIENT` in the original source.

**Comment by Electroid** [comment]
Make sure to open an issue before doing the programming -- because GS was removed intentionally, it's not something we want to support, really.

**Comment by TheMolkaPL** [comment]
> it's not something we want to support

What is the reason?

**Comment by Electroid** [comment]
None of the popular community maps use it, and it's implemented in a hacky, mostly hard-coded way.

**Comment by TheMolkaPL** [comment]
Okay, I'll maybe make an addon to PGM from it.

**Comment by Pablete1234** [comment]
@TheMolkaPL If you want pgm to support ghost squardon, consider adding individual modules or features to existing modules that add any needed & missing functionality for it. You'll probably also want an XML import with the classes and generic stuff for it.

**Pull request closed without merging**